samcli/commands/build/build_context.py

```diff
@@ -25,20 +26,20 @@ class BuildContext:
 
     def __init__(
         self,
-        resource_identifier,
-        template_file,
-        base_dir,
-        build_dir,
-        cache_dir,
-        cached,
-        mode,
-        manifest_path=None,
-        clean=False,
-        use_container=False,
-        parameter_overrides=None,
-        docker_network=None,
-        skip_pull_image=False,
-    ):
+        resource_identifier: Optional[str],
+        template_file: str,
+        base_dir: Optional[str],
+        build_dir: str,
+        cache_dir: str,
+        cached: bool,
+        mode: Optional[str],
+        manifest_path: Optional[str],
+        clean: bool,
+        use_container: bool,
+        parameter_overrides: dict,
+        docker_network: Optional[str],
+        skip_pull_image: bool,
```

Comment:
The main motivation to remove them is because `parameter_overrides`'s default value here is not correct. The default value is `{}`:
https://github.com/aws/aws-sam-cli/blob/eefcd0744fd90323da31232504b87449629e10eb/samcli/commands/_utils/options.py#L251-L260

So I remove them all default args here because the default value are defined in `@click...` somewhere already, defining here requires us to maintain two sources of truth. Also we always provide full arguments when invoking this funtion, so there is no need to have default arguments.